sled-agent/src/bin/sled-agent.rs

```diff
@@ -62,7 +62,7 @@ async fn do_run() -> Result<(), CmdError> {
                 rss_config_path.push("config-rss.toml");
                 rss_config_path
             };
-            let rss_config = rss_config_path.exists().then_some({
+            let rss_config = rss_config_path.exists().then(|| {
                 let rss_config =
                     RackInitializeRequest::from_file(rss_config_path)
                         .map_err(|e| CmdError::Failure(anyhow!(e)))?;
```

Comment:
I'm not sure this compiles - does this mean `rss_config` is now an `Option<Result<...>>`?

Reply:
> Actually, it didn't mean that; `then_some` was already returning an `Option<RackInitializeRequestParams>`; the `Result` was being destructured by the `?` on line 68. But you can't (conveniently) use `?` in a closure, so it did fail to compile.
> 
> It's possible one could omit the `?` and try to play games with `.transpose` or `.ok` or some combination of things like that, but I don't think the result would be comprehensible. My conclusion is that none of the slick methods on `bool` are really applicable here, so I just switched it back to an `if` expression.